src/chainstate/stacks/db/blocks.rs

```diff
@@ -3380,152 +3396,247 @@ impl StacksChainState {
+                                true
+                            } else {
+                                // not the first-ever block.  Does this connect to a previously-accepted
+                                // block in the headers database?
+                                let hdr_sql = "SELECT * FROM block_headers WHERE block_hash = ?1 AND consensus_hash = ?2".to_string();
+                                let hdr_args: &[&dyn ToSql] = &[
+                                    &candidate.parent_anchored_block_hash,
+                                    &candidate.parent_consensus_hash,
+                                ];
+                                let hdr_rows = query_rows::<StacksHeaderInfo, _>(
+                                    headers_conn,
+                                    &hdr_sql,
+                                    hdr_args,
+                                )
+                                .map_err(Error::DBError)?;
+
+                                match hdr_rows.len() {
+                                    0 => {
+                                        // no parent processed for this block
+                                        debug!(
+                                            "No such parent {}/{} for block, cannot process",
+                                            &candidate.parent_consensus_hash,
+                                            &candidate.parent_anchored_block_hash
+                                        );
+                                        false
+                                    }
+                                    1 => {
+                                        // can process this block
+                                        debug!(
+                                            "Have parent {}/{} for this block, will process",
+                                            &candidate.parent_consensus_hash,
+                                            &candidate.parent_anchored_block_hash
+                                        );
+                                        true
+                                    }
+                                    _ => {
+                                        // should be impossible -- stored the same block twice
+                                        unreachable!(
+                                            "Stored the same block twice: {}/{}",
+                                            &candidate.parent_anchored_block_hash,
+                                            &candidate.parent_consensus_hash
+                                        );
+                                    }
+                                }
+                            }
+                        };
+
+                        if can_attach {
+                            // try and load up this staging block and its microblocks
+                            match StacksChainState::load_staging_block(
```

Comment:
Is this loading the same block that was already loaded into `candidate`?

Reply:
Oh, I see, this function is necessary to finish loading the block's bytes

Reply:
No -- `load_staging_block` will load the block header and block data, whereas the SQL statement above only loads the block header.  I suppose I could remove the redundant block-header query by filling in `candidate.block_data` with:

```
                // load up associated block data
                staging_block.block_data =
                    StacksChainState::load_block_bytes(blocks_path, consensus_hash, block_hash)?
                        .unwrap_or(vec![]);
                Ok(Some(staging_block))
```

(which is what `load_staging_block` does).